### DISCUSSION TITLE
Unexpected errors when Java › Gradle › Build Server: Enabled is on

### DISCUSSION THREAD

**Discussion by gogenan** [microsoft/vscode-discussions#897] [Extension Development QnA]
When I build one of my Java/Gradle projects and Java › Gradle › Build Server: Enabled is on I see the following errors:
Output/ Build Server for Gradle

> Build starts at 2023-10-30 16:48:48
Start to build: classes testClasses
Task :compileJava started
Task :compileJava failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> invalid source release: 15

Problems

Archive for required library: '/auto/ns_nm/users/gennady/.gradle/caches/modules-2/files-2.1/com.github.cloudyrock.mongock/mongock-bom/4.3.8/c928882a2b20280fb100c02b778f98bed88c3626/mongock-bom-4.3.8.pom' in project 'mi-platform-manager' cannot be read or is not a valid ZIP file

However, the project build is successful from the command line and when building from the Gradle extension. Also, if I turn the Build Server off, the project is build successfully.
Can this issue be addressed by changing settings?

**Comment by theReynald** [comment]
Hi @gogenan,

The issue you're experiencing seems to stem from the integration of the Build Server protocol with your Java/Gradle project within Visual Studio Code. The error `invalid source release: 15` indicates that there is a mismatch between the Java SDK version expected by Gradle and the one being used by the build server. The second error about the library archive not being read or not being a valid ZIP file could be related to cache corruption or path issues.

Here are a few steps you could take to try and resolve these issues:

1. **Check Java SDK Configuration:**
   - Ensure that your `JAVA_HOME` environment variable is set to the correct JDK path for Java 15.
   - In Visual Studio Code settings, verify that the Java extension is configured to use the correct version of the JDK. Sometimes the extension may not pick up the correct JDK from `JAVA_HOME`.

2. **Gradle Settings:**
   - In your `build.gradle` file, make sure that the `sourceCompatibility` and `targetCompatibility` are set to Java 15.
   - Clear the Gradle cache by deleting the `.gradle` folder in your home directory or by using the command `gradle clean build --refresh-dependencies` to force a refresh of dependencies.

3. **VS Code Build Server:**
   - Since the build is successful when the Build Server is off, it might be an issue with the Build Server's configuration. You can check the settings related to the Java Build Server in Visual Studio Code and ensure it's using the correct Gradle version and settings.
   - Update your extensions to the latest versions to ensure that any fixes related to the Build Server are applied.

4. **Corrupted Archive:**
   - The error regarding the library archive could be due to a corrupted download. Manually clear out the cache directory mentioned in the error message, so Gradle can redownload the dependencies.
   - Verify if the path to the cached library is correct and accessible by the build server.

5. **VS Code Settings:**
   - If there are any custom settings in `.vscode/settings.json` related to the Java or Gradle extensions, review them to ensure they are not causing a conflict.

6. **Project Configuration:**
   - Ensure that your project's `.gradle` and `gradle` wrapper are correctly set up and that the wrapper is using the appropriate Gradle version.
   - Make sure that there are no local configuration files that might override the build settings when run inside Visual Studio Code.

If these steps do not resolve the issue, consider looking into the logs for the Java Language Server and the Gradle Build Server to get more detailed error information. These logs can sometimes provide more context on what's going wrong and can be found within the Output panel in VS Code when you set the output to the Java or Gradle tasks.

You might also want to check if there are any known issues or updates available for the Java extension pack or the Gradle extension in Visual Studio Code that could address this problem.

**Reply by gogenan** [reply]
Hi Reynald,

Regarding the Java sources issue:

1. I have several Gradle projects that are opened simultaneously in VS Code and use different JDKs. Therefore, I cannot use JAVA_HOME to resolve this.
2. As I've mentioned in my question, all my Gradle projects are successfully build from the command line. Therefore, the Gradle build files and wrapper allows them to use a proper JDK.
3. Again, as I've mentioned, all these projects are successfully built from the Gradle extension or when Gradle Build Server is disabled. I suppose that this is because that they work similarly to the command line.

Therefore, this seems as a bug in the Gradle Build Server, which fails to determine the proper JDK from the Gradle files of the project.

As for the second strange problem when a POM file is considered as a ZIP file, my comments above, except for JAVA_HOME, also apply to it.

There are no options of Gradle Build Server that allows any configuration related to the problem.

FYI, I've installed the latest update of VS Code and this did not help.

Version: 1.84.0
Commit: d037ac076cee195194f93ce6fe2bdfe2969cc82d
Date: 2023-11-01T11:28:21.782Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 3.10.0-1160.24.1.el7.x86_64

**Comment by burkeholland** [comment]
This is most likely an issue for the Java extension repo. Possibly here? https://github.com/microsoft/vscode-gradle/issues

As this appears to be a question not related to extension development, I’m closing this for now. Please feel free to re-open if you feel this is an error.
